In [2]:
import pandas as pd

In [3]:
df = pd.read_csv("Toronto_3cols.csv",index_col=0)

In [4]:
df.head()

PostalCode      Borough                            Neighborhood
0        M1B  Scarborough                          Rouge, Malvern
1        M1C  Scarborough  Highland Creek, Rouge Hill, Port Union
2        M1E  Scarborough       Guildwood, Morningside, West Hill
3        M1G  Scarborough                                  Woburn
4        M1H  Scarborough                               Cedarbrae

In [5]:
!pip install geocoder

    100% |████████████████████████████████| 102kB 19.5MB/s 


In [6]:
import geocoder # import geocoder

postalcodes=[]
latitudes=[]
longitudes=[]

RUN_GEOCODER = False

if RUN_GEOCODER:
    for postal_code in df.PostalCode:
        # initialize your variable to None
        lat_lng_coords = None
        # loop until you get the coordinates
        print(postal_code,end=' ')
        while(lat_lng_coords is None):        
            g = geocoder.google('{}, Toronto, Ontario'.format(postal_code))
            lat_lng_coords = g.latlng

        latitude = lat_lng_coords[0]
        longitude = lat_lng_coords[1]

        print(latitude,longitude)

        postalcodes.append(postal_code)
        latitudes.append(latitude)
        longitudes.append(longitude)

In [7]:
#!wget -O geospatial_data.csv http://cocl.us/Geospatial_data

The geocoder technique does not return data very well - could not get past the first postal code. So I am trying the csv file as mentioned in the project brief.

In [8]:
df_geo = pd.read_csv('geospatial_data.csv')
df_geo.head()

Postal Code   Latitude  Longitude
0         M1B  43.806686 -79.194353
1         M1C  43.784535 -79.160497
2         M1E  43.763573 -79.188711
3         M1G  43.770992 -79.216917
4         M1H  43.773136 -79.239476

In [10]:
df = df.set_index('PostalCode').join(df_geo.set_index('Postal Code')).reset_index()

In [11]:
df.to_csv("Toronto_wLatLong.csv")
df.head()

PostalCode      Borough                            Neighborhood   Latitude  \
0        M1B  Scarborough                          Rouge, Malvern  43.806686   
1        M1C  Scarborough  Highland Creek, Rouge Hill, Port Union  43.784535   
2        M1E  Scarborough       Guildwood, Morningside, West Hill  43.763573   
3        M1G  Scarborough                                  Woburn  43.770992   
4        M1H  Scarborough                               Cedarbrae  43.773136   

   Longitude  
0 -79.194353  
1 -79.160497  
2 -79.188711  
3 -79.216917  
4 -79.239476

In [12]:
df.shape

(103, 5)

We had 103 rows previously so this is good. 

In [13]:
df.describe(include='all')

PostalCode     Borough Neighborhood    Latitude   Longitude
count         103         103          103  103.000000  103.000000
unique        103          11          103         NaN         NaN
top           M4G  North York    Northwest         NaN         NaN
freq            1          24            1         NaN         NaN
mean          NaN         NaN          NaN   43.704608  -79.397153
std           NaN         NaN          NaN    0.052463    0.097146
min           NaN         NaN          NaN   43.602414  -79.615819
25%           NaN         NaN          NaN   43.660567  -79.464763
50%           NaN         NaN          NaN   43.696948  -79.388790
75%           NaN         NaN          NaN   43.745320  -79.340923
max           NaN         NaN          NaN   43.836125  -79.160497

Even better - no problematic values.

So let's check the info function.

In [14]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 103 entries, 0 to 102
Data columns (total 5 columns):
PostalCode      103 non-null object
Borough         103 non-null object
Neighborhood    103 non-null object
Latitude        103 non-null float64
Longitude       103 non-null float64
dtypes: float64(2), object(3)
memory usage: 4.1+ KB


And save for the next stage